subprojects/packagefiles/libusb/meson.build

```diff
@@ -0,0 +1,100 @@
+project('libusb', 'c', version: '1.0.26', meson_version: '>=0.60')
+
+add_project_arguments('-D_GNU_SOURCE', language: 'c')
+
+cc = meson.get_compiler('c')
+pconf = import('pkgconfig')
+
+thread_dep = dependency('threads')
+udev_dep = dependency('libudev', required: get_option('udev'))
+
+cdata = configuration_data()
+cdata.set('HAVE_CLOCK_GETTIME', cc.has_function('clock_gettime'))
+cdata.set('HAVE_EVENTFD', cc.has_function('eventfd'))
+cdata.set('HAVE_TIMERFD', cc.has_function('timerfd_create'))
+cdata.set('HAVE_SYSLOG', cc.has_function('syslog'))
+cdata.set('HAVE_PIPE2', cc.has_function('pipe2'))
+cdata.set('HAVE_PTHREAD_CONDATTR_SETCLOCK', cc.has_function('pthread_condattr_setclock'))
+cdata.set('HAVE_PTHREAD_SETNAME_NP', cc.has_function('pthread_setname_np'))
+cdata.set('HAVE_PTHREAD_THREADID_NP', cc.has_function('pthread_threadid_np'))
+
+cdata.set('HAVE_ASM_TYPES_H', cc.has_header('asm/types.h'))
+cdata.set('HAVE_NFDS_T', cc.has_header('nfds.h'))
+cdata.set('HAVE_SYS_TIME_H', cc.has_header('sys/time.h'))
+
+cdata.set('HAVE_STRUCT_TIMESPEC', cc.has_header_symbol('time.h', 'struct timespec'))
+
+incdirs = include_directories(['.', 'libusb', 'libusb/os'])
+install_headers('libusb/libusb.h', subdir: 'libusb-1.0')
+
+sources = files(
+  'libusb/core.c',
+  'libusb/descriptor.c',
+  'libusb/hotplug.c',
+  'libusb/io.c',
+  'libusb/strerror.c',
+  'libusb/sync.c',
+ )
+
+if host_machine.system() == 'windows'
+  cdata.set('PLATFORM_WINDOWS', true)
+  sources += files('libusb/os/events_windows.c', 'libusb/os/threads_windows.c')
+  if get_option('default_library') != 'static'
+    cdata.set('DEFAULT_VISIBILITY', '__declspec(dllexport)')
+  else
+    cdata.set('DEFAULT_VISIBILITY', '')
+  endif
+else
+  cdata.set('PLATFORM_POSIX', true)
+  cdata.set('DEFAULT_VISIBILITY', '__attribute__ ((visibility ("default")))')
+  sources += files('libusb/os/events_posix.c', 'libusb/os/threads_posix.c')
+endif
+
+if host_machine.system() == 'linux'
+  sources += 'libusb/os/linux_usbfs.c'
+  if udev_dep.found()
+    cdata.set('HAVE_LIBUDEV', true)
+    sources += 'libusb/os/linux_udev.c'
+  else
+    sources += 'libusb/os/linux_netlink.c'
+  endif
+elif host_machine.system() == 'haiku'
+  sources += files('libusb/os/haiku_usb_backend.cpp', 'libusb/os/haiku_pollfs.cpp', 'libusb/os/haiku_usb_raw.cpp')
+elif host_machine.system() == 'windows'
+  sources += files('libusb/os/windows_common.c', 'libusb/os/windows_usbdk.c', 'libusb/os/windows_winusb.c')
+elif host_machine.system() == 'darwin'
+  sources += files('libusb/os/darwin_usb.c')
+elif host_machine.system() == 'netbsd'
+  sources += files('libusb/os/netbsd_usb.c')
+elif host_machine.system() == 'openbsd'
+  sources += files('libusb/os/openbsd_usb.c')
+elif host_machine.system() == 'sunos'
+  sources += files('libusb/os/sunos_usb.c')
```

Comment:
A lot of this can be simplified as:
```
elif host_system in ['darwin', 'netbsd', 'openbsd', 'sunos']
    sources += files('libusb/os/@0@_usb.c'.format(host_system))
else
```